uefi_ovmf_64/kAFLAgentPkg/Include/Library/kAFLAgentLib.h

```diff
@@ -88,16 +102,40 @@ RunTestHarness (
 #define HYPERCALL_KAFL_USER_FAST_ACQUIRE	18
 /* 19 is already used for exit reason KVM_EXIT_KAFL_TOPA_MAIN_FULL */
 #define HYPERCALL_KAFL_USER_ABORT			20
-#define HYPERCALL_KAFL_TIMEOUT				21
+#define HYPERCALL_KAFL_TIMEOUT				21 /* deprecated */
+#define HYPERCALL_KAFL_RANGE_SUBMIT		29
+#define HYPERCALL_KAFL_REQ_STREAM_DATA		30
+#define HYPERCALL_KAFL_PANIC_EXTENDED		32
 
-#define PAYLOAD_SIZE						(128 << 10)				/* up to 128KB payloads */
-#define PROGRAM_SIZE						(128 << 20)				/* kAFL supports 128MB programm data */
-#define INFO_SIZE							(128 << 10)				/* 128KB info string */
+#define HYPERCALL_KAFL_CREATE_TMP_SNAPSHOT	33
+#define HYPERCALL_KAFL_DEBUG_TMP_SNAPSHOT	34 /* hypercall for debugging / development purposes */
 
-typedef struct{
+#define HYPERCALL_KAFL_GET_HOST_CONFIG		35
+#define HYPERCALL_KAFL_SET_AGENT_CONFIG		36
+
+#define HYPERCALL_KAFL_DUMP_FILE			37
+
+#define HYPERCALL_KAFL_REQ_STREAM_DATA_BULK 38
+#define HYPERCALL_KAFL_PERSIST_PAGE_PAST_SNAPSHOT 39
+
+/* hypertrash only hypercalls */
+#define HYPERTRASH_HYPERCALL_MASK			0xAA000000
+
+#define HYPERCALL_KAFL_NESTED_PREPARE		(0 | HYPERTRASH_HYPERCALL_MASK)
+#define HYPERCALL_KAFL_NESTED_CONFIG		(1 | HYPERTRASH_HYPERCALL_MASK)
+#define HYPERCALL_KAFL_NESTED_ACQUIRE		(2 | HYPERTRASH_HYPERCALL_MASK)
+#define HYPERCALL_KAFL_NESTED_RELEASE		(3 | HYPERTRASH_HYPERCALL_MASK)
+#define HYPERCALL_KAFL_NESTED_HPRINTF		(4 | HYPERTRASH_HYPERCALL_MASK)gre
```

Comment:
this is based on outdated nyx_api.h - can you rebase + copy?

Reply:
I found this in qemu repo

https://github.com/IntelLabs/kafl.qemu/blob/d407bab2549540140ba17301d08c6453aa989bc4/linux-headers/linux/kvm.h#L273-L276

Since I don't need nested virtualization there, I can maybe get rid of those lines ?

Reply:
You can get rid of them but ideally just minimize any differences. My comment was regarding line 128, the typo at the end of the line was fixed in latest targets/nyx.h